dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyTreeFormatter.java

```diff
@@ -61,15 +65,58 @@ private static void printDependencyTree(String coordinate)
    * Prints dependencies expressed in dependency paths in tree in similar way to mvn
    * dependency:tree.
    *
-   * @param dependencyPaths sorted dependency paths
+   * @param dependencyPaths dependency paths from @{@link
+   *     DependencyGraphBuilder#getCompleteDependencies(Artifact)}. Each element must have its
+   *     parent in the list, except the ones at the root.
    */
   public static String formatDependencyPaths(List<DependencyPath> dependencyPaths) {
     StringBuilder stringBuilder = new StringBuilder();
+    // Printing text representing a tree requires traversing the items in pre-order
+    ListMultimap<DependencyPath, DependencyPath> tree = buildDependencyPathTree(dependencyPaths);
+    // Empty dependency path is to retrieve children of root node
+    formatDependencyPathTreePreOrder(stringBuilder, tree, new DependencyPath());
+    return stringBuilder.toString();
+  }
+
+  private static void formatDependencyPathTreePreOrder(
+      StringBuilder stringBuilder,
+      ListMultimap<DependencyPath, DependencyPath> tree,
+      DependencyPath currentNode) {
+    String indentCharacter = "  ";
+    int depth = currentNode.size();
+    if (depth > 0) {
+      // Nodes at top have one or more depth
+      stringBuilder.append(Strings.repeat(indentCharacter, depth));
+      stringBuilder.append(currentNode.getLeaf());
+      stringBuilder.append("\n");
+    }
+    for (DependencyPath childPath : tree.get(currentNode)) {
+      formatDependencyPathTreePreOrder(stringBuilder, tree, childPath);
+    }
+  }
+
+  /**
+   * Builds ListMultiMap that represents a Maven dependency tree of parent-children relationship.
+   * The root node is represented as an empty {@link DependencyPath} and its children are
+   * the values for the root node. As Maven dependency is retrieved via BFS, the order of children
+   * matters.
+   *
+   * @param dependencyPaths a list of dependency path without assuming any order
```

Comment:
why "without assuming any order"? It's a list so it's ordered when called. Would the output be the same if the list were shuffled before this were called?

Reply:
Previous implementation of tree formatting was relying on the pre-order of `dependencyPaths`.



> Would the output be the same if the list were shuffled before this were called?

Yes, now it doesn't depend on the order of the item. Changed "List" to "Collection" to express this property as type.